### PR TITLE
Update to aravis 0.8.31

### PR DESCRIPTION
**Pull request by vsuryamurthy**
On top of the changes to the manually written files, I had to manually update some auto generated files as it was not compiling. Especially `aravis/src/auto/flags.rs` (a new file), I had to remove this ` #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]` from :

```
bitflags! {
        #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
        #[doc(alias = "ArvGvStreamOption")]
        pub struct GvStreamOption: u32 {
                #[doc(alias = "ARV_GV_STREAM_OPTION_NONE")]
                const NONE = ffi::ARV_GV_STREAM_OPTION_NONE as _;
                #[doc(alias = "ARV_GV_STREAM_OPTION_PACKET_SOCKET_DISABLED")]
                const PACKET_SOCKET_DISABLED = ffi::ARV_GV_STREAM_OPTION_PACKET_SOCKET_DISABLED as _;
        }
}
```

And also added a `clone()` to L76.

Edit: These above changes are no longer necessary with the latest glib version [0.20.0]. Yet to push those updates though as I am having a different issue.

### CODE REVIEW

**Comment by vsuryamurthy** [comment]
Also, the latest generated code also fixes this [issue](https://github.com/robohouse-delft/aravis-rs/issues/9). `Stream::push_buffer` no longer requires a borrow.

**Comment by vsuryamurthy** [comment]
`create_stream_full` in device seems to have been introduced to fix python issues (https://github.com/AravisProject/aravis/pull/714) so I added it to ignore list, not sure if it is actually necessary here.

File: aravis-sys/Cargo.toml [hunk 1]
```diff
@@ -24,34 +75,27 @@ v0_8_17 = ["v0_8_16"]
...
+workspace = true
+
 [build-dependencies]
 pkg-config = "0.3.27"
-system-deps = "6.0.5"
+system-deps = "7"
 
 [dev-dependencies]
-shell-words = "1.1.0"
+shell-words = "1.0.0"
```
Comment (de-vri-es) [inline aravis-sys/Cargo.toml]: Why downgrade this?

File: aravis-sys/Cargo.toml [hunk 2]
```diff
@@ -24,34 +75,27 @@ v0_8_17 = ["v0_8_16"]
...
+[dependencies.glib-sys]
+workspace = true
+
+[dependencies.gobject-sys]
+workspace = true
+
 [build-dependencies]
 pkg-config = "0.3.27"
-system-deps = "6.0.5"
+system-deps = "7"
```
Comment (de-vri-es) [inline aravis-sys/Cargo.toml]:
```suggestion
system-deps = "7.0.1"
```

**Review by de-vri-es — APPROVED** [review]

**Comment by de-vri-es** [comment]
Note to self: this PR contains breaking changes, so we need a proper version bump.

File: aravis-sys/Cargo.toml (hunk 1 above)
Comment (de-vri-es) [reply aravis-sys/Cargo.toml]:
```suggestion
shell-words = "1.1.0"
```

**Comment by de-vri-es** [comment]
Released as v0.10.0 :rocket:

Thanks for all the effort!

File: aravis-sys/Cargo.toml (hunk 1 above)
Comment (vsuryamurthy) [reply aravis-sys/Cargo.toml]: Ohh that was not intended!

**Pull request merged**